src/main/telemetry/smartport.c

```diff
@@ -352,15 +341,11 @@ bool canSendSmartPortTelemetry(void)
 
 void checkSmartPortTelemetryState(void)
 {
-    bool newTelemetryEnabledValue = (smartPortPortSharing == PORTSHARING_NOT_SHARED);
-    if (newTelemetryEnabledValue == smartPortTelemetryEnabled) {
+    if (!smartPortTelemetryEnabled) {
```

Comment:
Triggering `configureSmartPortTelemetryPort()` should probably happen directly in `initSmartPortTelemetry` (if possible).